smartapps/stj/mqtt-bridge.src/mqtt-bridge.groovy

```diff
@@ -524,10 +524,19 @@ def bridgeHandler(evt) {
         if (capability["attributes"].contains(json.type)) {
             settings[key].each {device ->
                 if (device.displayName == json.name) {
-                    if (capability.containsKey("action")) {
-                        def action = capability["action"]
-                        // Yes, this is calling the method dynamically
-                        "$action"(device, json.type, json.value)
+                    if (json.command == false) {
+                        if (device.getSupportedCommands().any {it.name == "setStatus"}) {
+                            log.debug "Setting state ${json.type} = ${json.value}"
+                            device.setStatus(json.type, json.value)
+                            state.ignoreEvent = json;
```

Comment:
Is there a race condition between this line and the line above?

Reply:
It shouldn't as they should happen in 2 different cycles. All of this event handler should happen in one cycle, then any changes to the device being reported should happen on the next, I could be wrong about this but I have been using both the garage opener and a light using this and haven't observed a race condition.